services/apps/src/apps/clients/iplant_groups.clj

```diff
@@ -1,14 +1,110 @@
 (ns apps.clients.iplant-groups
-  (:require [cemerick.url :as curl]
+  (:use [medley.core :only [remove-vals]]
+        [slingshot.slingshot :only [throw+]])
+  (:require [apps.util.config :as config]
+            [cemerick.url :as curl]
             [clj-http.client :as http]
-            [apps.util.config :as config]))
+            [clojure.string :as string]
+            [clojure-commons.exception :as cx]
+            [kameleon.uuids :refer [uuidify]]))
 
-(defn- lookup-subject-url
-  [short-username]
-  (str (curl/url (config/ipg-base) "subjects" short-username)))
+(def grouper-user "de_grouper")
+(def grouper-user-group-fmt "iplant:de:%s:users:de-users")
+(def grouper-app-permission-def-fmt "iplant:de:%s:apps:app-permission-def")
+(def grouper-app-resource-name-fmt "iplant:de:%s:apps:%s")
+
+(defn- grouper-user-group
+  []
+  (format grouper-user-group-fmt (config/env-name)))
+
+(defn- grouper-app-permission-def
+  []
+  (format grouper-app-permission-def-fmt (config/env-name)))
+
+(defn- grouper-app-resource-name
+  [app-id]
+  (format grouper-app-resource-name-fmt (config/env-name) app-id))
+
+(defn- grouper-url
+  [& components]
+  (str (apply curl/url (config/ipg-base) components)))
 
 (defn lookup-subject
   "Uses iplant-groups's subject lookup by ID endpoint to retrieve user details."
   [user short-username]
-  (-> (http/get (lookup-subject-url short-username) {:query-params {:user user} :as :json})
+  (-> (http/get (grouper-url "subjects" short-username) {:query-params {:user user} :as :json})
       (:body)))
+
+(defn load-app-permissions
+  "Loads app permissions from Grouper."
+  [user]
+  (let [id-from-resource #(last (string/split % #":"))]
+    (->> (http/get (grouper-url "attributes" "permissions")
+                   {:query-params {:user          grouper-user
+                                   :role          (grouper-user-group)
+                                   :attribute_def (grouper-app-permission-def)
+                                   :subject_id    user}
+                    :as           :json})
+         (:body)
+         (:assignments)
+         (group-by (comp uuidify id-from-resource :name :attribute_definition_name)))))
```

Comment:
Grouper returns one assignment for every permission level that the user has, so we group the permissions by app ID in order to make it easy to tell what permissions the user has for the app (or what actions the user can perform in Grouper parlance).